### PR TITLE
InteractingDoor event with KeycardThrow

### PR DESCRIPTION
**Pull request by IRacle1**
~~The only problem is that the interaction with the collision occurs several times, and the event is also invoked several times (an average of 3 times), but I do not think that this is a big problem~~
fixed

### CODE REVIEW

**Comment by louis1706** [comment]
I can look for make it not trigger multiple time

**Comment by louis1706** [comment]
> I can imagine the "trigger multiple times" would be a problem for plugins such as discord game logging plugins, which would send a log multiple times.

I already know how fixed it ^^ do not worry

**Review by louis1706 — APPROVED** [review]
it's all fine

File: Exiled.Events/EventArgs/ThrowKeycardInteractingEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    using Exiled.API.Features.Items;
+
+    using Interactables.Interobjects.DoorUtils;
+
+    using InventorySystem.Items.Keycards;
+
+    /// <summary>
+    /// Contains all informations before a keycard interacts with a door.
+    /// </summary>
+    public class ThrowKeycardInteractingEventArgs : EventArgs
```
Comment (NaoUnderscore) [inline Exiled.Events/EventArgs/ThrowKeycardInteractingEventArgs.cs:22]: I'd rename it to `ThrowingInteractableKeycard` or `KeycardInteracting`.

File: Exiled.Events/Patches/Events/Item/InteractingDoorWithThrow.cs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    using UnityEngine;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="KeycardPickup.ProcessCollision(Collision)"/>.
+    /// Adds the <see cref="Handlers.Player.InteractingDoor"/> event.
+    /// </summary>
+    [HarmonyPatch(typeof(KeycardPickup), nameof(KeycardPickup.ProcessCollision))]
+    internal static class InteractingDoorWithThrow
```
Comment (NaoUnderscore) [inline Exiled.Events/Patches/Events/Item/InteractingDoorWithThrow.cs:37]: Keep the same name as the event if possible.

File: Exiled.Events/Config.cs [hunk 3]
```diff
@@ -54,6 +54,12 @@ public sealed class Config : IConfig
         [Description("Indicates whether the blood can be spawned or not")]
         public bool CanSpawnBlood { get; set; } = true;
 
+        /// <summary>
+        /// Gets or sets a value indicating whether keycard throw can affect basic doors.
+        /// </summary>
+        [Description("Indicates whether keycard throw can affect on basic doors(door with empty permissions)")]
+        public bool CanKeycardThrowAffectOnBasicDoors { get; set; } = true;
```
Comment (NaoUnderscore) [inline Exiled.Events/Config.cs:61]: `CanKeycardThrowAffectDoors`

File: Exiled.Events/Config.cs [hunk 4]
```diff
@@ -54,6 +54,12 @@ public sealed class Config : IConfig
         [Description("Indicates whether the blood can be spawned or not")]
         public bool CanSpawnBlood { get; set; } = true;
 
+        /// <summary>
+        /// Gets or sets a value indicating whether keycard throw can affect basic doors.
+        /// </summary>
+        [Description("Indicates whether keycard throw can affect on basic doors(door with empty permissions)")]
```
Comment (NaoUnderscore) [inline Exiled.Events/Config.cs:60]: I'd change it to: _Indicates whether thrown keycards can affect doors that don't require any permissions_

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

**Comment by BuildBoy12** [comment]
While it does fire only once when the interaction is allowed, it still fires multiple times when it is not.

LczArmory Door:
image

Checkpoint:
image

**Comment by IRacle1** [comment]
my problem is logic, I'll fix it tomorrow

**Comment by louis1706** [comment]
> While it does fire only once when the interaction is allowed, it still fires multiple times when it is not.
> 
> LczArmory Door:
> image
> 
> Checkpoint:
> image
> 

Is not in base game already like that i don't see the problem with this ? It's pretty normal i thinks

**Comment by IRacle1** [comment]
It's still a duplicate event, lots of reasons why it's bad

**Comment by louis1706** [comment]
> It's still a duplicate event, lots of reasons why it's bad

I will try something later for see if it's do not break something

**Review by Jesus-QC — APPROVED** [review]

**Pull request merged**